### DISCUSSION TITLE
I'm struggling to write JavaScript with Zed.

### DISCUSSION THREAD

**Discussion by skatkov** [zed-industries/zed#18057] [Feature Requests]
Good day,

Thanks for amazing editor - I'm successfully writing ruby and Golang with it. But struggle to figure out how to write JavaScript at work with it and have to go back to sublime everytime I need to write JS.

Main friction as it seems, is that we use [semistandard](https://github.com/standard/semistandard). Zed supports prettier by default and it's great, it's just that using prettier instead of semistandard will not fly with my team.

I tried changing the configuration to use semistandard instead, but it doesn't get picked up. I assume, that I did something wrong, so creating a discussion instead (seems like a configuration issues, than a bug).

Did anyone had success at using anything other than prettier?

**Comment by bajrangCoder** [comment]
Zed allows the usage of external formatters through commands. You can configure this in settings.

- **Install the Formatter**: First, install the formatter globally using npm.
-  **Configure Zed Settings**:  you need to add your formatter's CLI command to Zed's settings for the specific language you want to format. Here's an example configuration for JavaScript:

   ```jsonc
   {
     "languages": {
       "JavaScript": {
         "formatter": {
           "external": {
             "command": "semistandard",
             "arguments": [] // Add any required arguments here
           }
         },
         "format_on_save": true
       }
     }
   }
   ```

Now, whenever you save a JavaScript file, Zed will use the specified formatter automatically.

Check out this : https://zed.dev/docs/configuring-languages#configuring-formatters

**Reply by skatkov** [reply]
Thanks for answer. But I did try already configuring a custom formatter this way, but unfortunately still it seems to be formatting according to prettier standards ('' turned into "" as example)

**Reply by bajrangCoder** [reply]
You can check `zed log` , as I am using a external formatter for `Blade` templating language and it's working.
Here is my config
```json
{
  "languages": {
    "Blade": {
      "formatter": {
        "external": {
          "command": "blade-formatter",
          "arguments": ["{buffer_path}"]
        }
      }
    }
  }
}
```

You can share your zed log after running the Formatter, it will in debugging.

**Reply by bajrangCoder** [reply]
In your case , pass --fix flag in argument to semistandard formatter as by default it checks and logs the errors.

```jsonc
{
  "languages": {
    "JavaScript": {
      "formatter": {
        "external": {
          "command": "semistandard",
          "arguments": ["--fix"] // don't pass buffer_path if you don't want to format a single file , as with current config it will format the whole project
        }
      }
    }
  }
}
```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
